src/wasm/wasm-binary.cpp

```diff
@@ -1691,9 +1691,11 @@ void WasmBinaryBuilder::processExpressions() {
           peek == BinaryConsts::Catch) {
         BYN_TRACE("== processExpressions finished with unreachable"
                   << std::endl);
-        readNextDebugLocation();
         lastSeparator = BinaryConsts::ASTNodes(peek);
-        pos++;
+        // Read the byte we peeked at. No new instruction is generated for it.
+        Expression* dummy;
```

Comment:
It looks like `readExpression` doesn't always store to its outparam, so this should be initialized to `nullptr` (I might be wrong, but doing so would still be more defensive).

Reply:
When does it not? I think it always does. But good idea regardless, I'll update.